app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/newdevicenotice/NewDeviceNoticeEmailAccessViewModel.kt

```diff
@@ -38,8 +47,24 @@ class NewDeviceNoticeEmailAccessViewModel @Inject constructor(
     }
 
     private fun handleContinueClick() {
-        // TODO PM-8217: update new device notice status and navigate accordingly
-        sendEvent(NavigateToTwoFactorOptions)
+        if (state.isEmailAccessEnabled) {
+            val displayStatus =
+                if (featureFlagManager.getFeatureFlag(FlagKey.NewDevicePermanentDismiss)) {
+                    CAN_ACCESS_EMAIL_PERMANENT
+                } else {
+                    CAN_ACCESS_EMAIL
+                }
+
+            authRepository.setNewDeviceNoticeState(
+                NewDeviceNoticeState(
+                    displayStatus = displayStatus,
+                    delayDate = ZonedDateTime.now(),
```

Comment:
🤔 Maybe I'm misunderstanding the views/flows, but shouldn't `delayDate` be `nil` here given the status is not `HAS_SEEN`?

Reply:
You are right the date here is irrelevant